### ISSUE TITLE
Support differentiability of multi-argument metric tensor - or conciously decide against it

### ISSUE THREAD

**Issue by dwierichs** [PennyLaneAI/pennylane#1947]
### Feature details

Currently, the metric tensor of a QNode (with `hybrid=True`) when computed with respect to multiple arguments returns a tuple of metric tensors, one for each argument, and not a single full tensor spanning all arguments. This is currently the intended behaviour.
However, it prevents differentiability, as tuple-output functions are not supported in multiple interfaces.

The output of the differentiation with respect to multiple arguments will probably be in turn multi-dimensional tuples, which is a very inconvenient "array" type to process further. It might therefore be a bad idea to support this overall and one could require users to either 
1. wrap their QNodes such that a single argument captures all argument w.r.t which the metric is to be computed.
or 
2. compute the metric tensor with `hybrid=False` and account for classical processing manually instead of via the `hybrid=True` functionality.

### Implementation

I am not sure whether there is a convenient way of implementing this or whether it has to be done simply manually by differentiating each of the output tensors. This would probably be a quite general utility that enables differentiation of tuples.

### How important would you say this feature is?

1: Not important. Would be nice to have.

### Additional information

This is mostly a request for a discussion to have.

**Comment by dwierichs** [comment]
> the same must be done,

I'm not sure I follow this phrasing. In JAX it "simply works", doesn't it?

I notice, though, that for example in JAX, `jax.hessian(fn, argnums=[0,1])` of a scalar function `fn` with two input arguments returns a nested tuple, so generalizing the tuple-shaped output of a Jacobian to multi-dimensional tuples is something adopted in JAX. (And yes, the Jacobian of the Hessian is a 3d tuple... )

In `torch`, the Hessian output also is a 2D tuple but it can not be differentiated being a tuple-valued function.

**Comment by josh146** [comment]
Oh, I get it now @dwierichs! That is, the automatic nesting of tuples when asking for the Jacobian/Hessian.

**Comment by dwierichs** [comment]
I think this is a non-issue by now, in particular as the new return type system encourages handling of tuples rather than automatically stacking tensors. Do you agree, @josh146 ?

**Comment by josh146** [comment]
Agree!

**Comment by dwierichs** [comment]
Closing because of the above.
